src/diffusers/models/modeling_utils.py

```diff
@@ -623,6 +655,51 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
             user_agent=user_agent,
             **kwargs,
         )
+        # no in-place modification of the original config.
+        config = copy.deepcopy(config)
```

Comment:
Doing this because we modify the config with quantization config and in any case, that fails, we should be able to retrieve the original model config back before the failure happened. This is my reasoning. WDYT?